source/kraken/tests/apply_disruption_test.cpp

```diff
@@ -109,3 +114,508 @@ BOOST_FIXTURE_TEST_CASE(simple_train_cancellation, SimpleDataset) {
     // but the vp should be equals again
     BOOST_CHECK_EQUAL(vj->base_validity_pattern(), vj->rt_validity_pattern());
 }
+
+/*
+ * small helper to check VJs
+ */
+void check_vj(const nt::VehicleJourney* vj, const std::vector<nt::StopTime>& sts) {
+    BOOST_REQUIRE_EQUAL(vj->stop_time_list.size(), sts.size());
+    for (size_t idx = 0; idx < sts.size(); ++idx) {
+        BOOST_CHECK_EQUAL(vj->stop_time_list[idx].departure_time, sts[idx].departure_time);
+        BOOST_CHECK_EQUAL(vj->stop_time_list[idx].arrival_time, sts[idx].arrival_time);
+        BOOST_CHECK_EQUAL(vj->stop_time_list[idx].stop_point, sts[idx].stop_point);
+    }
+}
+
+/*
+ *
+ * test to close different stop point on different hours
+ *
+ *
+ *                            closed         closed
+ *                            [10-12]        [11-14]
+ *                S1 ---------- S2 ------------S3
+ *
+ * vj1:           8h            9h             10h
+ * no vj1_adapted
+ *
+ * vj2:           10h           10h30          11h
+ * vj2_adapted:   10h           X              X         TODO: With the current algo, some VJs won't circulate at all
+ *
+ * vj3:           10h                          11h
+ * vj3_adapted:   10h                          X
+ *
+ * vj4:           8h            10h            10h30
+ * vj4_adapted:   8h            X              10h30
+ *
+ * vj5:           8h            13h            13h30
+ * vj5_adapted:   8h            13h            X
+ *
```

Comment:
maybe we can add a vj6: not s1 10h30 11h to check if we remove all the stop times.